code/modules/reagents/withdrawal/generic_addiction.dm

```diff
@@ -0,0 +1,256 @@
+	. = ..()
+	affected_carbon.apply_status_effect(/datum/status_effect/woozy)
+
+/datum/addiction/stimulants/withdrawal_enters_stage_3(mob/living/carbon/affected_carbon)
+	. = ..()
+	affected_carbon.add_movespeed_modifier(MOVESPEED_ID_STIMULANTS, TRUE, 100, override=TRUE, multiplicative_slowdown = 1)
+
+/datum/addiction/stimulants/end_withdrawal(mob/living/carbon/affected_carbon)
+	. = ..()
+	affected_carbon.remove_actionspeed_modifier(ACTIONSPEED_ID_STIMULANTS)
+	affected_carbon.remove_status_effect(/datum/status_effect/woozy)
+	affected_carbon.remove_movespeed_modifier(MOVESPEED_ID_STIMULANTS)
+
+///Alcohol
+/datum/addiction/alcohol
+	name = "alcohol"
+	withdrawal_stage_messages = list("I could use a drink...", "Maybe the bar is still open?..", "God I need a drink!")
+
+/datum/addiction/alcohol/withdrawal_stage_1_process(mob/living/carbon/affected_carbon, delta_time)
+	. = ..()
+	affected_carbon.Jitter(5 * delta_time)
+
+/datum/addiction/alcohol/withdrawal_stage_2_process(mob/living/carbon/affected_carbon, delta_time)
+	. = ..()
+	affected_carbon.Jitter(10 * delta_time)
+	affected_carbon.hallucination = max(5 SECONDS, affected_carbon.hallucination)
+
+/datum/addiction/alcohol/withdrawal_stage_3_process(mob/living/carbon/affected_carbon, delta_time)
+	. = ..()
+	affected_carbon.Jitter(15 * delta_time)
+	affected_carbon.hallucination = max(5 SECONDS, affected_carbon.hallucination)
+	if(DT_PROB(4, delta_time))
+		affected_carbon.apply_status_effect(/datum/status_effect/seizure)
+
+/datum/addiction/hallucinogens
+	name = "hallucinogen"
+	withdrawal_stage_messages = list("I feel so empty...", "I wonder what the machine elves are up to?..", "I need to see the beautiful colors again!!")
+
+/datum/addiction/hallucinogens/withdrawal_enters_stage_2(mob/living/carbon/affected_carbon)
+	. = ..()
+	var/atom/movable/plane_master_controller/game_plane_master_controller = affected_carbon.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+	game_plane_master_controller.add_filter("hallucinogen_wave", 10, wave_filter(300, 300, 3, 0, WAVE_SIDEWAYS))
+	game_plane_master_controller.add_filter("hallucinogen_blur", 10, angular_blur_filter(0, 0, 3))
+
+
+/datum/addiction/hallucinogens/withdrawal_enters_stage_3(mob/living/carbon/affected_carbon)
+	. = ..()
+	affected_carbon.apply_status_effect(/datum/status_effect/trance, 40 SECONDS, TRUE)
+
+/datum/addiction/hallucinogens/end_withdrawal(mob/living/carbon/affected_carbon)
+	. = ..()
+	var/atom/movable/plane_master_controller/game_plane_master_controller = affected_carbon.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+	game_plane_master_controller.remove_filter("hallucinogen_blur")
+	game_plane_master_controller.remove_filter("hallucinogen_wave")
+	affected_carbon.remove_status_effect(/datum/status_effect/trance, 40 SECONDS, TRUE)
+
+/datum/addiction/maintenance_drugs
+	name = "maintenance drug"
+	withdrawal_stage_messages = list("", "", "")
```

Comment:
this should really be an empty list or null, and the messenger should handle that case